src/server.rs

```diff
@@ -326,59 +327,87 @@ impl Handler {
     ///
     /// When the shutdown signal is received, the connection is processed until
     /// it reaches a safe state, at which point it is terminated.
-    #[instrument(skip(self))]
+    #[instrument(
+        level = "info",
+        name = "Handler::run",
+        skip(self),
+        fields(
+            ?peer_addr = self.connection.peer_addr().unwrap(),
```

Comment:
the question mark goes before the expression, not the field name. also, we should probably use the `Display` impl for `SocketAddr` rather than `Debug`:
```suggestion
            peer_addr = %self.connection.peer_addr().unwrap(),
```